src/test/ui/suggestions/unnamable-types.rs

```diff
@@ -0,0 +1,23 @@
+// Test that we do not suggest to add type annotations for unnamable types.
```

Comment:
Please add a test cases for function defs and generators.

Reply:
Fixed. I have added two more cases at the end of this file, one for a function definition, and one for a generator.